tests/components/nibe_heatpump/test_climate.py

```diff
@@ -44,15 +56,249 @@ async def test_basic(
     coils[climate.current] = 20.5
     coils[climate.setpoint_heat] = 21.0
     coils[climate.setpoint_cool] = 30.0
-    coils[climate.mixing_valve_state] = "ON"
+    coils[climate.mixing_valve_state] = 20
     coils[climate.use_room_sensor] = "ON"
-    coils[unit.prio] = "HEAT"
+    coils[unit.prio] = "OFF"
     coils[unit.cooling_with_room_sensor] = "ON"
 
     await async_add_model(hass, model)
 
-    assert hass.states.get(entity_id) == snapshot(name="1. initial")
+    assert hass.states.get(entity_id) == snapshot(name="initial")
+
+    mock_connection.mock_coil_update(unit.prio, "COOLING")
+    assert hass.states.get(entity_id) == snapshot(name="cooling")
+
+    mock_connection.mock_coil_update(unit.prio, "HEAT")
+    assert hass.states.get(entity_id) == snapshot(name="heating")
+
+    mock_connection.mock_coil_update(climate.mixing_valve_state, 30)
+    assert hass.states.get(entity_id) == snapshot(name="idle (mixing valve)")
+
+    mock_connection.mock_coil_update(climate.mixing_valve_state, 20)
+    mock_connection.mock_coil_update(unit.cooling_with_room_sensor, "OFF")
+    assert hass.states.get(entity_id) == snapshot(name="heating (only)")
+
+    mock_connection.mock_coil_update(climate.use_room_sensor, "OFF")
+    assert hass.states.get(entity_id) == snapshot(name="heating (auto)")
+
+    mock_connection.mock_coil_update(unit.prio, None)
+    assert hass.states.get(entity_id) == snapshot(name="off (auto)")
+
+    if climate.active_accessory:
+        mock_connection.mock_coil_update(climate.active_accessory, "OFF")
+        assert hass.states.get(entity_id) == snapshot(
+            name="unavailable (not supported)"
+        )
+
+    coils.clear()
+    assert hass.states.get(entity_id) == snapshot(name="unavailable")
+
+
+@pytest.mark.parametrize(
+    ("model", "climate_id", "entity_id"),
+    [
+        (Model.S320, "s1", "climate.climate_system_s1"),
+        (Model.F1155, "s2", "climate.climate_system_s2"),
+    ],
+)
+async def test_set_temperature(
+    hass: HomeAssistant,
+    mock_connection: MockConnection,
+    model: Model,
+    climate_id: str,
+    entity_id: str,
+    coils: dict[int, Any],
+    entity_registry_enabled_by_default: None,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test setting temperature."""
+    climate = CLIMATE_COILGROUPS[model.series][climate_id]
+    unit = UNIT_COILGROUPS[model.series]["main"]
+    if climate.active_accessory is not None:
+        coils[climate.active_accessory] = "ON"
+    coils[climate.current] = 20.5
+    coils[climate.setpoint_heat] = 21.0
+    coils[climate.setpoint_cool] = 30.0
+    coils[climate.mixing_valve_state] = 20
+    coils[climate.use_room_sensor] = "ON"
+    coils[unit.prio] = "OFF"
+    coils[unit.cooling_with_room_sensor] = "ON"
+
+    await async_add_model(hass, model)
```

Comment:
Looks the same as in the test above. Could we use a fixture or a function for it?